doc/build.txt

```diff
@@ -122,18 +159,43 @@ New build systems can be added to |g:build#systems|:
   \ {
   \   'my-build-system':
   \   {
-  \     'file'    : 'foo.json',
-  \     'command' : 'foomake',
-  \     'init'    : './configure',
+  \     'file'     : 'foo.json',
+  \     'commands' : { 
+  \       'do'   : 'foomake',
+  \       'init' : 'mkdir .foomake_cache',
+  \       'build': 'foomake build',
+  \       'clean': 'foomake wipe',
+  \       'run'  : 'foomake cmd',
+  \       'bar'  : 'foomake bar', " an uncommon command
+  \     },
   \   },
   \ }
 <
-Every build system must have a name, a file entry and a command entry.
-The file entry contains a comma-separated list of filenames, describing the
-build files to search for. The command entry contains the command for
-running the build system. It will be run in the directory containing the
-build file. The init entry is optional and contains the command for
-initializing the build.
+
+Every build system must have a name, a file entry and a commands entry.  The
+file entry contains a comma-separated list of filenames or globs (see vim help
+for |glob()|), describing the build files to search for. The commands entry
+contains the commands for running the build system. It will be run in the
+directory containing the build file.
+You never have to add `info` or `help` that are handled internally.
+You should provide `init` if it makes sense for your build system.
+You can add new commands if you want, they will be recognized on a per-build
+system basis.
+You can omit some of the common commands if they don't make sense in your case.
+
+Every command will be executed in the directory of the detected build file and can contain
+four different placeholders:
+
+                                                                   *build_%PATH%*
```

Comment:
Add a tag to this section so they can be referenced in the documentation of `g:build#languages.`.